src/main/java/org/candlepin/model/EntitlementCurator.java

```diff
@@ -1301,4 +1301,17 @@ private Predicate ilike(CriteriaBuilder cb, Expression<String> expression, Strin
         Predicate[] array = new Predicate[predicates.size()];
         return predicates.toArray(array);
     }
+
+    public List<String> getProductContentUUIDsByEntitlementId(String entitlementId) {
+        String sql = "SELECT c.uuid FROM cp_entitlement e " +
+            "JOIN cp_pool p ON p.id = e.pool_id " +
+            "JOIN cp2_product_content pc ON pc.product_uuid = p.product_uuid " +
```

Comment:
in a production scenario, this would never return any data, because the main product that is referenced on the pool (p.product_uuid) would never actually have any content itself. it is only engineering products that have contents.

I would suggest keeping this query as is, but also add a second query to this method, which instead of just this join, has 2 joins, to make sure it links content that are hanging on provided products:
```
JOIN cp2_product_provided_products pp ON p.product_uuid=pp.product_uuid
JOIN cp2_product_content pc ON pp.provided_product_uuid=pc.product_uuid
```

Reply:
Sure, Added the both type of queries within same method.
Shouldn't we write spec test closer to production scenario ? Lot of the spec test have the content attached to main product.

Reply:
Currently for this specific query, there are no guarding against IN operator. I'll add that once I have more clarity on how do we want to split these queries.

Reply:
As discussed, we want to keep both of these queries. Added the Unit test for this method.